OpenSim/Tools/MarkerPlacer.cpp

```diff
@@ -242,6 +243,47 @@ bool MarkerPlacer::processModel(Model* aModel,
     /* Load the static pose marker file, and average all the
     * frames in the user-specified time range.
     */
+    TimeSeriesTableVec3 staticPoseTable{aPathToSubject + _markerFileName};
+    const auto& timeCol = staticPoseTable.getIndependentColumn();
+    auto numRowsInRange = std::count_if(timeCol.cbegin(),
+                                        timeCol.cend(),
+                                        [&] (const double& val) {
+                                            return val >= _timeRange[0] &&
+                                                   val <= _timeRange[1];
+                                        });
+    const auto avgRow = staticPoseTable.averageRow(_timeRange[0],
+                                                   _timeRange[1]);
+    for(int r = staticPoseTable.getNumRows() - 1; r >= 0; --r) {
+        if(staticPoseTable.getIndependentColumn()[r] >= _timeRange[0] &&
+           staticPoseTable.getIndependentColumn()[r] <= _timeRange[1]) {
+            if(numRowsInRange > 1) {
+                staticPoseTable.removeRowAtIndex(r);
+                --numRowsInRange;
+            } else {
+                staticPoseTable.updRowAtIndex(r) = avgRow;
+            }
+        }
+    }
+    OPENSIM_THROW_IF(!staticPoseTable.hasTableMetaDataKey("Units"),
+                     Exception,
+                     "MarkerPlacer::processModel -- Marker file does not have "
+                     "'Units'.");
+    Units
+    staticPoseUnits{staticPoseTable.getTableMetaData<std::string>("Units")};
+    double scaleFactor = staticPoseUnits.convertTo(aModel->getLengthUnits());
+    OPENSIM_THROW_IF(SimTK::isNaN(scaleFactor),
+                     Exception,
+                     "Model has unspecified units.");
+    if(std::fabs(scaleFactor - 1) >= SimTK::Eps) {
+        for(unsigned r = 0; r < staticPoseTable.getNumRows(); ++r)
+            staticPoseTable.updRowAtIndex(r) *= scaleFactor;
+
+        staticPoseUnits = aModel->getLengthUnits();
+        staticPoseTable.removeTableMetaDataKey("Units");
+        staticPoseTable.addTableMetaData("Units",
+                                         staticPoseUnits.getAbbreviation());
+    }
+    
     MarkerData* staticPose = new MarkerData(aPathToSubject + _markerFileName);
     staticPose->averageFrames(_maxMarkerMovement, _timeRange[0], _timeRange[1]);
     staticPose->convertToUnits(aModel->getLengthUnits());
```

Comment:
Is the sticking point using `TimeSeriesTable_<Vec3>::averageRow()` as a replacement for `MarkerData::averageFrame()`? If this is more complex than anticipated there should be a note or issue filed briefly describing the problem.

Also, the average frame and conversion of units will only apply to the `MarkerData` pose that is never used, since the `staticPoseTable` is used to create the `MarkersReference` used by the Solver. So effectively, these calls have no impact on the algorithm. It seems like we are lucky that the first row in the `staticPoseTable` is sufficiently close to the average frame of the `staticPose` (`MarkerData`) to still pass the tests.

The future PR should have a test that `Table::averageRow()` yields the single frame result of the removed `MarkerData::averageFrame()`.